### PR TITLE
Add Switch component

### PR DESCRIPTION
**Pull request by philearley**
# Description
A new component which effectively restyles the Checkbox component.

# Screenshots / Gifs

### Checked

screen shot 2018-08-08 at 14 57 58

### Unchecked
screen shot 2018-08-08 at 14 58 01

### Loading checked
loading-on

### Loading unchecked
loading-off

### Focus states
screen shot 2018-08-09 at 17 01 07
screen shot 2018-08-09 at 17 00 56

# Testing Instructions
Pull the branch down locally.
`$npm install`
`$npm start`
Navigate to localhost:8095
Go to the Switch component in the components list

### CODE REVIEW

**Comment by andrewjtait** [comment]
Please make sure this is cross browser tested.

**Comment by philearley** [comment]
At desk demoed with @lukeemmerson 

Tested in Internet Explorer, Chrome, Firefox, Safari, mobile Safari.
Issue found with missing the first loading dot.  This was an issue with classname and is now resolved. Retested each browser again to validate.

Testing consisted of:
Selecting the label and selecting the switch.
Enabling and disabling through props.
Validated while loading is true you cannot click the component.

File: src/components/switch/switch.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+function renderIcon(icon, loading) {
+  if (loading) {
+    return (
+      <div className='carbon-loading-dots'>
+        <div className='bounce bounce1' />
+        <div className='bounce bounce2' />
+        <div className='bounce bounce3' />
+      </div>
```
Comment (ianoxley) [inline src/components/switch/switch.js]: Is it worth moving the loading dots to their own component? i.e. could they be used elsewhere, or do they only really make sense as part of the `Switch` component?

**Comment by philearley** [comment]
At desk tested the focus state with @harpalsingh during implementation and validated with @lukeemmerson

**Pull request merged**